ldes-server-port-fetch/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fetching/services/CompactionCandidateComparator.java

```diff
@@ -0,0 +1,60 @@
+package be.vlaanderen.informatievlaanderen.ldes.server.fetching.services;
+
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.model.LdesFragmentIdentifier;
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.model.TreeRelation;
+import be.vlaanderen.informatievlaanderen.ldes.server.fetching.entities.CompactionCandidate;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static be.vlaanderen.informatievlaanderen.ldes.server.fragmentation.services.FragmentComparator.hasNoConnections;
+
+public class CompactionCandidateComparator {
```

Comment:
Can we merge this class and FragmentComparator into one class? The below logic is quite complex and I would be surprised if there aren't any bugs in it so it is likely they will need adjustments. Having the logic in two classes increases the risk of only fixing the bugs in one of the two.

Reply:
Although the idea behind it is a good one, in practicallity, this would mean that the CompactionCandidate entity would need to be brought to the fragmentisers-common, which is not really ideal.
To keep this separation of modules, these classes need to be kept separate